Update service rules from Emily M's review (#380)
DFE-Digital/register-early-career-teachers-public: documentation/service-rules/school-admins.md

We tell them they can update the email at a later point. This is because we know sometimes school users register ECTs in advance, when their school email address might not be ready yet.

> We check the email address given and make sure it doesn’t exist for an ongoing ECT or mentor record with a different TRN. This would mean any emails that are attached to an open `ect_at_school_period` or open '`mentor_at_school_period`. This is because we shouldn’t have email addresses that are the same for entirely different people, when both are are still undergoing training.

Reviewer: ```suggestion
We check the email address given and make sure it doesn’t exist for an ongoing ECT or mentor record with a different TRN. This would mean any emails that are attached to an open `ect_at_school_period` or open `mentor_at_school_period`. This is because we shouldn’t have email addresses that are the same for entirely different people, when both are are still undergoing training.
```

Registration is only finished when an ECT is also given a mentor.

> All ECTs need mentors, it's part of the support they should receive at every school. ECTs can start training without having a mentor assigned on the Register ECTs service, and lead providers will get funded for the ECT's training. However, mentors still need to be assigned to ECTs so they can:

Reviewer: ```suggestion
All ECTs need mentors, it’s part of the support they should receive at every school. ECTs can start training without having a mentor assigned on the Register ECTs service, and lead providers will get funded for the ECT’s training. However, mentors still need to be assigned to ECTs so they can:
```

Regardless of where the journey to assign a mentor starts, schools are taken to a question where they are asked if they’re assigning a mentor they’ve already registered, or registering a new mentor.

This will show all existing mentors that are still actively at the school - they have an open `mentor_at_school_period`.

If the school selects a previously registered mentor, they will successfully end the journey to assign a mentor, and also finish registering their ECT. This will change the ECT’s status from ‘mentor required’ to ‘registered’. 

If the school registers a new mentor, they’re taken to the questions outlined below.

> We would not allow here or later when they find a mentor's record for an ECT to be assigned as their own mentor.

Reviewer: ```suggestion
We would not allow here or later when they find a mentor’s record for an ECT to be assigned as their own mentor.
```

The school user is asked for that mentor’s email address.

> We check the email address given and make sure it doesn’t exist for an ongoing ECT or mentor record with a different TRN. This would mean any emails that are attached to an open `ect_at_school_period` or open '`mentor_at_school_period`. This is because we shouldn’t have email addresses that are the same for entirely different people, when both are are still undergoing training.

Reviewer: ```suggestion
We check the email address given and make sure it doesn’t exist for an ongoing ECT or mentor record with a different TRN. This would mean any emails that are attached to an open `ect_at_school_period` or open `mentor_at_school_period`. This is because we shouldn’t have email addresses that are the same for entirely different people, when both are are still undergoing training.
```
